### PR TITLE
S3 cache Deployment - Tiler

### PR DESCRIPTION
**Pull request by Rub21**
- Keep statefulset pod to work with a persistent disk (for this case we have the serve and cache cleaner in the same pod)
- For Cache in s3 bucket , the template will create two deployments one for tiler server and other for cache cleaner. 
- Adding Max_connection for DB, in case we have may tiler deployment running.

cc @geohacker @batpad

### CODE REVIEW

File: images/tiler-db/config/update_postgresql.sh [hunk 1]
```diff
@@ -0,0 +1,7 @@
+#!/bin/sh
+set -e
+if [ "${POSTGRES_DB_MAX_CONNECTIONS}X" != "X" ]; then
+  set -x;
+  sed -i -e"s/^.*max_connections =.*$/max_connections = $POSTGRES_DB_MAX_CONNECTIONS/" $PGDATA/postgresql.conf
```
Comment (geohacker) [inline images/tiler-db/config/update_postgresql.sh:5]: This is potentially dangerous and will have impact on memory consumption — so we should evaluate whether we add more memory or increase shmax. @Rub21 @batpad

File: images/tiler-server/Dockerfile [hunk 2]
```diff
@@ -50,4 +50,4 @@ COPY ./start.sh .
 COPY ./expire-watcher.sh .
 COPY ./seed-by-diffs.sh .
 COPY ./tile_cache_downloader.sh .
-CMD ./start.sh & ./tile_cache_downloader.sh & ./expire-watcher.sh
+# CMD ./start.sh & ./tile_cache_downloader.sh & ./expire-watcher.sh
```
Comment (geohacker) [inline images/tiler-server/Dockerfile]: This shouldn't be commented out right?

Comment (Rub21) [reply images/tiler-server/Dockerfile]: @geohacker yes, you are right, it should not be applied to oms-seed, for ohm  the tiler server and cache cleaner are running in different  pods, and we need to pass different  the starting commands.

**Review by geohacker — APPROVED** [review]

File: images/tiler-db/config/update_postgresql.sh [hunk 3]
```diff
@@ -0,0 +1,26 @@
+#!/bin/sh
+
+# max_connections
+if [[ "${POSTGRES_DB_MAX_CONNECTIONS}X" != "X" ]]; then
```
Comment (batpad) [inline images/tiler-db/config/update_postgresql.sh:4]: this if statement looks a bit weird to me? Why not just check if POSTGRES_DB_MAX_CONNECTIONS is set or so?

File: osm-seed/values.yaml [hunk 4]
```diff
@@ -61,7 +61,7 @@ db:
   enabled: true
   image:
     name: 'developmentseed/osmseed-db'
-    tag: '0.1.0-n377.hd099819'
+    tag: '0.1.0-n423.hdb62b0d'
```
Comment (batpad) [inline osm-seed/values.yaml:64]: we really should avoid committing these changes to the repo. these should all ideally just be something like `chartpress_replace` in the repository?

**Review by batpad — APPROVED** [review]

File: images/tiler-db/config/update_postgresql.sh (hunk 1 above)
Comment (Rub21) [reply images/tiler-db/config/update_postgresql.sh:5]: @geohacker  we are using this for DB tuning,  it is necessary to be there, or is there any other way that we can change those values?

File: images/tiler-db/config/update_postgresql.sh (hunk 3 above)
Comment (Rub21) [reply images/tiler-db/config/update_postgresql.sh:4]: It check if the env Var`POSTGRES_DB_MAX_CONNECTIONS` was set

File: osm-seed/values.yaml (hunk 4 above)
Comment (Rub21) [reply osm-seed/values.yaml:64]: Looks like it was set  when i ran `chartpress` 😬 , Do  you want me to remove that?

**Pull request merged**